apps/frontend/src/pages/access-management/libs/components/group-users-table/group-users-table.tsx

```diff
@@ -62,33 +75,47 @@ const GroupUsersTable = ({
 	const error = errors["userIds"]?.message;
 	const hasError = Boolean(error);
 
-	if (
+	const hasUser = userData.length !== EMPTY_LENGTH;
+
+	const isLoading =
 		usersDataStatus === DataStatus.IDLE ||
-		usersDataStatus === DataStatus.PENDING
-	) {
-		return <Loader />;
-	}
+		(usersDataStatus === DataStatus.PENDING && !hasUser);
 
 	return (
 		<>
-			<span className={styles["table-title"]}>Users</span>
-			<div className={styles["group-users-table"]}>
-				<Table<UserRow>
-					columns={userColumns}
-					data={userData}
-					getRowId={getRowId}
-					onRowSelect={onToggle}
-					selectedRowIds={selectedUserIds}
-				/>
-				<TablePagination
-					background="secondary"
-					onPageChange={onPageChange}
-					onPageSizeChange={onPageSizeChange}
-					page={page}
-					pageSize={pageSize}
-					totalItemsCount={usersTotalCount}
-				/>
+			<div>
+				<span className={styles["table-title"]}>Users</span>
+				<GroupUsersSearch onChange={handleSearchChange} />
 			</div>
+			{isLoading ? (
+				<Loader />
+			) : (
+				<div className={styles["group-users-table"]}>
+					{hasUser ? (
+						<>
+							<Table<UserRow>
+								columns={userColumns}
+								data={userData}
+								getRowId={getRowId}
+								onRowSelect={onToggle}
+								selectedRowIds={selectedUserIds}
+							/>
+							<TablePagination
+								background="secondary"
+								onPageChange={onPageChange}
+								onPageSizeChange={onPageSizeChange}
+								page={page}
+								pageSize={pageSize}
+								totalItemsCount={usersTotalCount}
+							/>
+						</>
+					) : (
+						<p className={styles["empty-placeholder"]}>
+							No users found matching your search criteria.
+						</p>
```

Comment:
we already have empty placeholder for table, maybe instead we can pass this text to it?